nk/main.go

```diff
@@ -244,19 +246,37 @@ func createVanityKey(keyType, vanity, entropy string, max int) nkeys.KeyPair {
 		log.Fatalf("Can not generate base32 encoded strings to match '%s'", vanity)
 	}
 
-	for i := 0; i < max; i++ {
-		spin := spinners[i%len(spinners)]
-		fmt.Fprintf(os.Stderr, "\r\033[mcomputing\033[m %s ", string(spin))
-		kp := genKeyPair(pre, entropy)
-		pub, _ := kp.PublicKey()
-		if strings.HasPrefix(pub[1:], vanity) {
-			fmt.Fprintf(os.Stderr, "\r")
-			return kp
-		}
+	var found nkeys.KeyPair
+	var wg sync.WaitGroup
+	cpu := runtime.NumCPU()
+	wg.Add(cpu)
+	for i:=0; i < cpu; i++ {
+		go func() {
+			for i := 0; i < max; i++ {
```

Comment:
Should max be divided up by `cpu`?

Reply:
if they are concurrent we are giving more chance to find it, but maybe that is not intuitive.

Reply:
Agree but I think max should still be honored.